test/test_gvl_timing.rb

```diff
@@ -87,4 +87,49 @@ def test_measure_and_inspect
     expected = "#<GVLTiming::Timer total=0.10s running=0.00s idle=0.10s stalled=0.00s yields=1>"
     assert_equal expected, timer.inspect
   end
+
+  def test_current_state
+    timer = GVLTiming::Timer.new
+    timer.start
+
+    idle_thread = Thread.new do
+      sleep 0.05
+      timer.current_state
+    end
+
+    sleep 0.1
+
+    state_while_sleeping = idle_thread.join.value
+
+    assert_equal :idle, state_while_sleeping
+    assert_equal :running, timer.current_state
```

Comment:
The `prev_state` in the C world is `current_state` in the Ruby world, because we use the `prev_state` value after the hook gets called while we use the `current_state` method in the Ruby world between state changes.